core/providers/huggingface/huggingface.go

```diff
@@ -0,0 +1,990 @@
+			rawResponse, bifrostErr := providerUtils.HandleProviderResponse(body, &huggingfaceAPIResponse, providerUtils.ShouldSendBackRawResponse(ctx, provider.sendBackRawResponse))
+			if bifrostErr != nil {
+				resultsChan <- providerResult{provider: inferProvider, err: bifrostErr}
+				return
+			}
+
+			var rawRespMap map[string]interface{}
+			if rawResponse != nil {
+				if converted, ok := rawResponse.(map[string]interface{}); ok {
+					rawRespMap = converted
+				}
+			}
+
+			resultsChan <- providerResult{
+				provider: inferProvider,
+				response: &huggingfaceAPIResponse,
+				latency:  latency.Milliseconds(),
+				rawResp:  rawRespMap,
+			}
+		}(infProvider)
+	}
+
+	// Close results channel after all goroutines complete
+	go func() {
+		wg.Wait()
+		close(resultsChan)
+	}()
+
+	// Aggregate results
+	aggregatedResponse := &schemas.BifrostListModelsResponse{
+		Data: make([]schemas.Model, 0),
+	}
+	var totalLatency int64
+	var successCount int
+	var firstError *schemas.BifrostError
+	var rawResponses []map[string]interface{}
+
+	for result := range resultsChan {
+		if result.err != nil {
+			if firstError == nil {
+				firstError = result.err
+			}
+			continue
+		}
+
+		if result.response != nil {
+			providerResponse := result.response.ToBifrostListModelsResponse(providerName, result.provider)
+			if providerResponse != nil {
+				aggregatedResponse.Data = append(aggregatedResponse.Data, providerResponse.Data...)
+				totalLatency += result.latency
+				successCount++
+				if result.rawResp != nil {
+					rawResponses = append(rawResponses, result.rawResp)
+				}
+			}
+		}
+	}
+
+	// If all requests failed, return the first error
+	if successCount == 0 && firstError != nil {
+		return nil, firstError
+	}
+
+	// Calculate average latency
+	if successCount > 0 {
+		aggregatedResponse.ExtraFields.Latency = totalLatency / int64(successCount)
+	}
+
+	if providerUtils.ShouldSendBackRawResponse(ctx, provider.sendBackRawResponse) && len(rawResponses) > 0 {
+		// Combine all raw responses into a single map
+		combinedRaw := make(map[string]interface{})
+		for i, raw := range rawResponses {
+			combinedRaw[fmt.Sprintf("provider_%d", i)] = raw
+		}
+		aggregatedResponse.ExtraFields.RawResponse = combinedRaw
+	}
+
+	return aggregatedResponse, nil
+}
+
+// ListModels queries the Hugging Face model hub API to list models served by the inference provider.
+func (provider *HuggingFaceProvider) ListModels(ctx context.Context, keys []schemas.Key, request *schemas.BifrostListModelsRequest) (*schemas.BifrostListModelsResponse, *schemas.BifrostError) {
+
+	if err := providerUtils.CheckOperationAllowed(schemas.HuggingFace, provider.customProviderConfig, schemas.ListModelsRequest); err != nil {
+		return nil, err
+	}
+	if provider.customProviderConfig != nil && provider.customProviderConfig.IsKeyLess {
+		return provider.listModelsByKey(ctx, schemas.Key{}, request)
+	}
+	return providerUtils.HandleMultipleListModelsRequests(
```

Comment:
we dont need multiple list models handler here right? cause we are using non-auth list models i suppose?